### PR TITLE
[native] Added slide updating based on prop change on Android

### PR DESCRIPTION
**Pull request by tomwalters**
When updating the index on Android previously the slider wouldn't update the selected page. This code adds this feature, as well as adding a new prop of 'animateTransitions' which leverages the ViewPagerAndroid API methods for animating/jumping. Previously reported under [#45](https://github.com/oliviertassinari/react-swipeable-views/issues/45).

TODO: Implement the animating for iOS based on the new prop.

### CODE REVIEW

File: src/index.native.scroll.js [hunk 1]
```diff
@@ -105,13 +114,24 @@ class SwipeableViews extends Component {
...
+    const indexHasChanged = (typeof index === 'number' && index !== this.props.index);
+
+    if (indexHasChanged) {
       this.setState({
         indexLatest: index,
         offset: {
           x: this.state.viewWidth * index,
         },
       }, () => {
+        if (indexHasChanged) {
```
Comment (oliviertassinari) [inline src/index.native.scroll.js]: This check seems useless.

File: src/index.native.scroll.js [hunk 2]
```diff
@@ -35,6 +35,14 @@ const styles = StyleSheet.create({
 class SwipeableViews extends Component {
   static propTypes = {
     /**
+     * If true, changes to the index prop will cause an
+     * animated transition, otherwise it'll just jump to
+     * the new slide.
+     *
+     * TODO: Implement this on iOS
+     */
+    animateTransitions: React.PropTypes.bool,
```
Comment (oliviertassinari) [inline src/index.native.scroll.js]: Are you using `animateTransitions={false}`? If not, I would just remove it 😁 .

**Comment by oliviertassinari** [comment]
@tomwalters That's going in the right direction 👍.

File: src/index.native.scroll.js (hunk 2 above)
Comment (tomwalters) [inline src/index.native.scroll.js]: Yep, check line 128

Comment (oliviertassinari) [inline src/index.native.scroll.js]: No, I mean. Do you have a use case for this property?

File: src/index.native.scroll.js (hunk 1 above)
Comment (tomwalters) [inline src/index.native.scroll.js]: Yep good point, as it's already within the if. This is what you get for coding late in the day😅

File: src/index.native.scroll.js (hunk 2 above)
Comment (tomwalters) [inline src/index.native.scroll.js]: So in my case: I have infinite scroll implemented with slide 0 representing the last slide in my deck. When i get to it I want to jump to the end without animation

Comment (oliviertassinari) [inline src/index.native.scroll.js]: I see. Let's add it then 👍.

**Pull request merged**